lab_04

pearvitin opened the pull request.

pearvitin committed:
lab_04

pearvitin commented:
В task2 неправильные пояснения под "#". Из объяснения не понятно, нужно ли "по умолчанию" всегда удалять последний элемент, или только в случаях, когда index не подается на входе в функцию. Понимаю, как исполнить оба случая, но прислал тот, что проходит проверку на курсе.

pearvitin committed:
Update and rename lask_03_lab_04.py to lask_2_lab_04.py

pearvitin committed:
Rename task_02_lab_04.py to task_03_lab_04.py

pearvitin committed:
Rename lask_2_lab_04.py to lask_02_lab_04.py

pearvitin committed:
Rename lask_02_lab_04.py to lask_3_lab_04.py

pearvitin committed:
Rename task_03_lab_04.py to task_02_lab_04.py

pearvitin committed:
Rename lask_3_lab_04.py to lask_03_lab_04.py

pearvitin committed:
Update task_02_lab_04.py

pearvitin committed:
Update task_02_lab_04.py

EgorOb commented on lab_04/task_02_lab_04.py:
```diff
@@ -0,0 +1,30 @@
+    # str_ = sorted(str_) если требуется сортировка
+    str_ = "".join(c for c in str_ if c.isalpha()).lower()  # нижний регистр, зачистка
+    counter = 0
+    for letter in str_:
```
for уже есть в 10 строке, не рационально использовать ещё один цикл для того, чтобы просто записать данные в словарь. В данной задаче лучше не делать словарь `slowar` глобальной переменной так как при повторной инициализации функции будет использоваться тот де словарь, а не новый. 
Попробуйте просто всё провести в одном цикле for и приведение к нижнему регистру и проверку является ли этот символ буквой.  Создавать новую отчищенную строку нет смысла, так как она нигде больше не используется.

EgorOb commented on lab_04/lask_03_lab_04.py:
```diff
@@ -0,0 +1,14 @@
+# TODO реализовать функцию delete удаления элемента из списка по индексу
+def delete(list_, index=None):
+    #  Почему нельзя просто параметрах написать index=-1?
```
Чтобы усложнить жизнь и поработали с обработкой типа None

EgorOb commented on lab_04/lask_03_lab_04.py:
```diff
@@ -0,0 +1,14 @@
+def delete(list_, index=None):
+    #  Почему нельзя просто параметрах написать index=-1?
+    if index is None:
+        list_ = list_[:-1]
```
Так как это функция и если вы больше нигде это не будете использовать, чтобы запоминать это значение, то можно сразу использовать return
`return list_[:-1]`

EgorOb commented on lab_04/lask_03_lab_04.py:
```diff
@@ -0,0 +1,14 @@
+        list_ = list_[:-1]
+    else:
+        list_ = list_[0:index] + list_[index+1:]
+    return list_
```
И тут вместо 6-8 строки
`return list_[:index] + list_[index+1:]`

EgorOb reviewed (approved):
Чтобы получить полный балл то, пожалуйста, перепишите функцию get_count_char, чтобы решала всё через один цикл и без ненужной предварительной очистки. Очистку можно делать прямо для символа.

pearvitin committed:
ver2

ушел от 2 циклов 'for'.
slowar занес в функцию, но теперь не могу его принять во вторую функцию.
В задаче не требуется выводить словарь с процентами.
Можно реализовать, оставив slowar глобальной переменной, но очищать его каждый раз при входе в первую функцию?

EgorOb reviewed (approved):
Зачтено. Вы можете просто этот же файл загрузить (но уже измененный) прямо в свою ветку (в данном случае lab_04). Аналогично как грузили данные файлы. Заходите в папку lab_04  в ветке lab_04 нажимаете загрузить файлы, кидаете измененный файл task_02_lab_04.py проверяете, что внизу коммит пойдет именно в вашу ветку lab_04 (по умолчанию оно будет коммитить в ту ветку с которой заливали, оно сейчас нам и нужно), новую ветку создавать не нужно, как в случае когда коммитите для создания пулл реквеста. Нажимаете зеленую кнопку, заходите в пулл реквест и скидываете мне ссылку.
